includes/KSES.php

```diff
@@ -26,20 +26,46 @@
 
 namespace Google\Web_Stories;
 
+use WP_Post_Type;
+
 /**
  * KSES class.
  *
  * Provides KSES utility methods to override the ones from core.
  */
 class KSES {
 	/**
-	 * Initializes KSES filters for stories.
+	 * Initializes KSES filters for stories only if capability check are met.
 	 *
 	 * @since 1.0.0
 	 *
 	 * @return void
 	 */
 	public function init() {
+
+		$create_posts     = false;
+		$post_type_object = get_post_type_object( Story_Post_Type::POST_TYPE_SLUG );
+		if (
+			$post_type_object instanceof WP_Post_Type &&
+			property_exists( $post_type_object->cap, 'create_posts' )
+		) {
+
+			$create_posts = current_user_can( $post_type_object->cap->create_posts );
```

Comment:
Would `$post_type_object->cap->edit_posts` be more correct?

Reply:
This is what I did originally and it didn't work. `create_posts` maps to `edit_web-stories`, which both sub and author user levels have. 
https://github.com/google/web-stories-wp/blob/42ddafa62692caaeca3dd0859733e2363912fd7a/includes/Story_Post_Type.php#L265
https://github.com/google/web-stories-wp/blob/42ddafa62692caaeca3dd0859733e2363912fd7a/includes/Story_Post_Type.php#L257

Reply:
So said mapping is done here:

https://github.com/WordPress/wordpress-develop/blob/8d7e7d03ff8ad036678851f959792f76d9759025/src/wp-includes/post.php#L1624-L1627

If it maps to that, and both roles have it, why doesn't the check for it work then? 🤔 

Certainly `$post_type_object->cap->create_posts === $post_type_object->cap->edit_posts` in this case.

Reply:
For some reason that didn't work, but just tried again and seems to work. Changed in https://github.com/google/web-stories-wp/pull/6077/commits/4714be3059a6260c0fe274f4e3d24a93b01c3073